### ISSUE TITLE
Callback to update learning rate does not update on

### ISSUE THREAD

**Issue by SwamiKannan** [DLR-RM/stable-baselines3#1336]
### 🐛 Bug

I am trying to write a callback that monitors the steps completed and the rewards received in a custom environment. The main intention is that when the episode completes a specific number of steps or achieves a specific value of reward, the callback changes the learning_rate of the model. However, while all the print statements show up indicating that the conditional statements are being entered to, the model's learning rate itself does not get updated.

I have also tried using get_parameters and set_parameters in _on_step() as follows:

```
def _on_step(self):
        self.params=self.model.get_parameters()
        self.lr=self.params['policy.optimizer']['param_groups'][0]['lr']
        self.call_rew=self.training_env.get_attr('reward')[0]
        if self.n_calls % self.check_freq == 0:
            print('Entering steps switch')
            self.lr=min(self.lr*self.lr_steps_mult,1e-5)
            print('New lr params',self.lr)
            self.params['policy.optimizer']['param_groups'][0]['lr']=self.lr
            print('New lr params',self.params['policy.optimizer']['param_groups'][0]['lr'])
            self.model.set_parameters(self.params)
        elif self.call_rew%5==0 and self.call_rew>0:
            print('Entering rewards switch with total_reward',self.call_rew)
            self.lr=min(self.lr*lr_rew_mult,1e-5)
            print('New lr params',self.lr)
            self.model.set_parameters(self.params)
        return True
```

I have seen one similar issue in the Issue repository but that doesn't seem to be working now. Could be a version issue. 

Thanks for all the help in advance.

### Code example

from stable_baselines3.common.callbacks import BaseCallback

class LearningRateCallback(BaseCallback):

    def __init__(self, freq=1000,lr_steps_mult=0.999,lr_rew_mult=0.999,verbose=1):
        super().__init__(verbose)
        self.check_freq=freq
        self.lr_steps_mult=lr_steps_mult
        self.lr_rew_mult=lr_rew_mult

    def _on_step(self):
        self.call_rew=self.training_env.get_attr('reward')[0]
        if self.n_calls % self.check_freq == 0:
            print('Entering steps switch')
            self.lr=self.model.learning_rate
            self.lr=min(self.lr*self.lr_steps_mult,1e-5)
            print('New lr params',self.lr)
            self.model.learning_rate=self.lr
        elif self.call_rew%5==0 and self.call_rew>0:
            print('Entering rewards switch with total_reward',self.call_rew)
            self.lr=self.model.learning_rate
            self.lr=min(self.lr*self.lr_steps_mult,1e-5)
            print('New lr params',self.lr)
            self.model.learning_rate=self.lr
        return True

### Relevant log output / Error message

```shell
All the print statements show up:

Initiating A2C training
Logging to DinoEnv\A2C_25000\logs\A2C_5
Entering steps switch
New lr params 9.502544225688347e-06

However, in the verbose statement, it still shows as:
------------------------------------
| rollout/              |          |
|    ep_len_mean        | 17.2     |
|    ep_rew_mean        | -33.8    |
| time/                 |          |
|    fps                | 1        |
|    iterations         | 100      |
|    time_elapsed       | 420      |
|    total_timesteps    | 500      |
| train/                |          |
|    entropy_loss       | -0       |
|    explained_variance | -53.8    |
|    learning_rate      | 0.001    |
|    n_updates          | 15417    |
|    policy_loss        | -0       |
|    value_loss         | 51.6     |
------------------------------------
```

### System Info

Installation: pip install stable-baselines3
GPU: NVidia RTX 3070
Python version 3.9.12
Pytorch version: 1.13.0
gym version: 0.21.0

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo
- [X] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/)
- [X] I have provided a minimal working example to reproduce the bug
- [X] I have checked my env using the env checker
- [X] I've used the [markdown code blocks](https://help.github.com/en/articles/creating-and-highlighting-code-blocks) for both code and stack traces.

**Comment by araffin** [comment]
Duplicate of https://github.com/DLR-RM/stable-baselines3/issues/856 https://github.com/DLR-RM/stable-baselines3/issues/735#issuecomment-1047638011 and https://github.com/DLR-RM/stable-baselines3/issues/942

**Comment by SwamiKannan** [comment]
Thanks for your assistance @araffin. The one comment that I had referred to in my post was #856 . I had applied that but hadn't worked out. I should have been clearer about that. 
#735 refers to change in learning rate independent of n_steps. I was specifically working on n_steps
